File: Exercise/Medium/07. Sort and Filter Array/Code/Sort and Filter Array.py
import sys

def sort_and_filter(arr):
    count5 = 0
    count6 = 0
    count7 = 0

    for num in arr:
        if num == 5:
            count5 += 1
        elif num == 6:
            count6 += 1
        elif num == 7:
            count7 += 1

    for i in range(count5):
        print("5", end=" ")
    for i in range(count6):
        print("6", end=" ")
    for i in range(count7):
        print("7", end=" ")
    print()

def main():
    line = input()

    arr = []
    ss = line.split()
    for num in ss:
        num = int(num)
        if num >= 5 and num <= 7:
            arr.append(num)

    sort_and_filter(arr)
    return 0

if __name__ == "__main__":
    sys.exit(main())

File: Exercise/Medium/07. Sort and Filter Array/Code/Sort and Filter Array.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void sortAndFilter(List<Integer> arr) {
        int count5 = 0, count6 = 0, count7 = 0;
        for (int num : arr) {
            if (num == 5) count5++;
            else if (num == 6) count6++;
            else if (num == 7) count7++;
        }

        for (int i = 0; i < count5; i++) System.out.print("5 ");
        for (int i = 0; i < count6; i++) System.out.print("6 ");
        for (int i = 0; i < count7; i++) System.out.print("7 ");
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        String line = scanner.nextLine();
        
        String[] tokens = line.split(" ");
        List<Integer> arr = new ArrayList<>();
        for (String token : tokens) {
            int num = Integer.parseInt(token.trim());
            if (num >= 5 && num <= 7) {
                arr.add(num);
            }
        }

        sortAndFilter(arr);
    }
}


